pretyping/patternops.ml

```diff
@@ -294,7 +294,7 @@ let rec subst_pattern
   | PInt _
   | PFloat _
   | PString _ -> pat
-  | PUninstantiated (PGenarg g) -> PUninstantiated (PGenarg (Genarg.generic_substitute subst g))
+  | PUninstantiated (PGenarg g) -> PUninstantiated (PGenarg (Gensubst.generic_substitute subst g))
```

Comment:
this is a bugfix as all the registers are on gensubst and not the duplicated (removed in this PR) API in genarg.ml